### PR TITLE
nginx: enable test for http.endpoint tag

### PR DESCRIPTION
**Pull request by cataphract**
## Motivation

Enables the http.endpoint tag tests. See https://github.com/DataDog/nginx-datadog/pull/267 and https://github.com/DataDog/dd-trace-cpp/pull/260

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed and the CI green, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

File: tests/test_resource_renaming.py [hunk 1]
```diff
@@ -51,7 +51,7 @@ def setup_http_endpoint_edge_cases(self):
 
     def test_http_endpoint_edge_cases(self):
         """Test that edge cases are handled correctly"""
-        assert get_endpoint_tag(self.r_long_path) == "/resource_renaming/a/b/c/d/e/f/g"
+        assert get_endpoint_tag(self.r_long_path) in ("/resource_renaming/a/b/c/d/e/f/g", "/resource_renaming/a/b/c/d/e/f/g/")
```
Comment (cataphract) [inline tests/test_resource_renaming.py]: RFC-1051 is ambiguous on whether truncated paths should have a trailing slash.

File: utils/build/docker/cpp_nginx/nginx/nginx.conf [hunk 2]
```diff
@@ -18,6 +18,11 @@ http {
           try_files /hello.html =404;
         }
 
+        location ~ ^/resource_renaming/ {
```
[the remark this answers is not in the document]
Comment (cataphract) [reply utils/build/docker/cpp_nginx/nginx/nginx.conf:21]: not strictly required for the tests to pass, since the RFC has no carve out for 404 responses, but advice followed

**Review by cbeauchesne — APPROVED** [review]
